cmake/modules/FindFFMPEG.cmake

```diff
@@ -0,0 +1,88 @@
+# Distributed under the MIT License.
+# Copyright (c) 2022 sakertooth <sakertooth@gmail.com>
+
+#[=======================================================================[.rst:
+FindFFMPEG
+-------
+
+Finds the FFmpeg library.
+
+Result Variables
+^^^^^^^^^^^^^^^^
+
+This will define the following variables:
+
+``FFMPEG_FOUND``
+  True if the system has found the required FFmpeg components.
+
+``FFMPEG_INCLUDE_DIRS``
+  The include directories of the required FFmpeg components.
+
+``FFMPEG_LIBRARIES``
+  The libraries of the required FFmpeg components.
+
+
+Cache Variables
+^^^^^^^^^^^^^^^
+
+For each FFmpeg component:
+  avcodec,
+  avformat,
+  avutil,
+  avfilter,
+  avdevice,
+  swresample,
+  swscale
+
+The following cache variables may also be set:
+
+``FFMPEG_<component>_INCLUDE_DIR``
+  The include directory for the FFmpeg component.
+
+``FFMPEG_<component>_LIBRARY``
+  The library for the FFmpeg component.
+
+#]=======================================================================]
+
+find_package(PkgConfig)
+set(_possibleComponents avcodec avformat avutil avfilter avdevice swresample swscale)
+
+foreach(component ${FFMPEG_FIND_COMPONENTS})
+  string(TOUPPER ${component} _upperComponent)
+  pkg_check_modules(PC_${_upperComponent} lib${component})
```

Comment:
Note that pkg-config isn't a mandatory dependency, so might be worth running only if `PKG_CONFIG_FOUND`.